spanner/client.go

```diff
@@ -456,6 +456,38 @@ func (c *Client) ReadWriteTransaction(ctx context.Context, f func(context.Contex
 	return ts, err
 }
 
+// BeginReadWriteTransaction starts a read-write transaction. Commit() or
+// Rollback() should be called when ending a transaction. If Commit() or
+// Rollback() is not called properly, then it can cause an leaked session that
+// is borrowed from the pool but is not returned.
+//
+// This method should only be used if a custom error handling is needed. For
+// normal use cases, client.ReadWriteTransaction should be used because it has
+// robust error handlings and retries.
+func (c *Client) BeginReadWriteTransaction(ctx context.Context) (*ReadWriteTransaction, error) {
```

Comment:
This method should return something else than `*ReadWriteTransaction`. It can probably be a wrapper around the current `ReadWriteTransaction` struct, but we should not re-use the existing struct directly for this purpose, as we are also adding a `Commit()` and `Rollback()` method. These methods must **not** be called when using the normal `Client.ReadWriteTransaction` method, but will be visible to all users, and that will be very confusing.